File: HowFastIsThis/lazyPrime.py
import time


def main():
    max = 100000
    primes = []
    for x in range (2, max):
        prime = True
        for y in range(2, int(x**0.5) + 1):
            if x % y == 0:
                prime = False
                break
        if prime:
            primes.append(x)
    return primes

if __name__ == "__main__":

    startTime = time.perf_counter()
    primes = main()
    endTime = time.perf_counter()

    elapsed = endTime - startTime
    print(f"Found {len(primes)} primes in {elapsed} seconds")

File: HowFastIsThis/lazyPrime.go
package main

import (
	"fmt"
	"time"
)

func main() {
	startTime := time.Now()
	max := 100000
	primeSlice := []int{}
	for x := 0; x < max; x++ { 
		prime := true
		for y := 2; y < x -1; y ++ {
			if x % y == 0 {
				prime = false
				break
			}
		}
		if prime == true {
			primeSlice = append(primeSlice, x)
		}
	}
	
	elapsed := time.Since(startTime) //.Since, Nice!

	fmt.Printf(fmt.Sprintf("identified %d primes in %s seconds", len(primeSlice), elapsed))

}
